lib/service/storage/virtualIndex.ts

```diff
@@ -0,0 +1,157 @@
+import Crypto from "crypto";
+
+import { Service } from "../service";
+
+export class VirtualIndex extends Service {
+  public static createEvent = "core:storage:virtualindex:cache:create";
+  public static deleteEvent = "core:storage:virtualindex:cache:delete";
+
+  public static createVirtualIndexEvent =
+    "core:storage:virtualindex:create:after";
+
+  public static deleteVirtualIndexEvent =
+    "core:storage:virtualindex:delete:after";
+
+  constructor() {
+    super("VirtualIndex", global.kuzzle.config.services.storageEngine);
+  }
+
+  public softTenant: Map<string, string> = new Map(); //Key : virtual index, value : real index //TODO rename!
+
+  async init(): Promise<void> {
+    await this.buildCollection();
+    await this.initVirtualTenantList();
```

Comment:
The methods in this class should not use the term `tenant`

Reply:
done